sda-download/internal/storage/storage.go

```diff
@@ -140,6 +167,15 @@ type S3Conf struct {
 	NonExistRetryTime time.Duration
 }
 
+func newS3SeekableBackend(config S3Conf) (*s3SeekableBackend, error) {
+	sb, err := newS3Backend(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &s3SeekableBackend{*sb}, nil
+}
+
```

Comment:
Is a dedicated backed really needed?
It is after all just a duplication of the existing backend.

Reply:
Correct.

I made it as a separate to be able to possibly merge it but still use the previous functionality only if you do not feel confident with it (e.g. your use cases don't care about delivering parts and you prefer higher performance). Can also be imlemented with a parameter to control behaviour of the driver.